### PR TITLE
[ITVBTCC] Update playlist extractor (closes #25240)

### LINKED ISSUE #25240: itv.com/btcc Unable to find videos
## Checklist

- [x] I'm reporting a broken site support
- [x] I've verified that I'm running youtube-dl version **2020.05.08**
- [x] I've checked that all provided URLs are alive and playable in a browser
- [x] I've checked that all URLs and arguments with special characters are properly quoted or escaped
- [x] I've searched the bugtracker for similar issues including closed ones

## Verbose log

```
[debug] System config: []
[debug] User config: ['-f', 'bestvideo[ext=mp4][height<=1080]+bestaudio[ext=m4a]/best[ext=mp4][height<=1080]/best[height<=1080]/best', '--console-title']
[debug] Custom config: []
[debug] Command-line args: ['https://www.itv.com/btcc/articles/btcc-2019-brands-hatch-gp-race-action', '-v']
[debug] Encodings: locale cp1252, fs mbcs, out cp850, pref cp1252
[debug] youtube-dl version 2020.05.08
[debug] Python version 3.4.4 (CPython) - Windows-10-10.0.18362
[debug] exe versions: ffmpeg git-2019-12-19-99f505d, ffprobe git-2019-12-19-99f505d
[debug] Proxy map: {}
[ITVBTCC] btcc-2019-brands-hatch-gp-race-action: Downloading webpage
[download] Downloading playlist: BTCC 2019: Brands Hatch GP race action
[ITVBTCC] playlist BTCC 2019: Brands Hatch GP race action: Collected 0 video ids (downloading 0 of them)
[download] Finished downloading playlist: BTCC 2019: Brands Hatch GP race action
```

## Description

Youtube-dl seems unable to find videos on any of the pages I have tried from https://www.itv.com/btcc/btcc-2019-races recently. I was able to download from itv.com/btcc links I had in a batch file around January but the page format seems to have changed from itv.com/btcc/races/[page name] to itv.com/btcc/articles/[page name] since then.

### PR DESCRIPTION
**Pull request by franhp**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/ytdl-org/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Bug fix
- [ ] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

ITV BTCC seems to have changed their platform and the way they expose their playlists from directly into the HTML into creating a JSON with all the necessary data. The old URLs still work and redirect to the new ways, so I've added backward compatibility just in case.

### CODE REVIEW

File: youtube_dl/extractor/itv.py [hunk 1]
```diff
@@ -305,7 +315,7 @@ def _real_extract(self, url):
                     'referrer': url,
                 }),
                 ie=BrightcoveNewIE.ie_key(), video_id=video_id)
-            for video_id in re.findall(r'data-video-id=["\'](\d+)', webpage)]
+            for video_id in re.findall(r'["\']data["\']:{["\']id["\']:(\d+),', webpage)]
```
Comment (dstftw) [inline youtube_dl/extractor/itv.py:318]: Do not remove the old pattern.

File: youtube_dl/extractor/itv.py [hunk 2]
```diff
@@ -278,15 +278,25 @@ def extract_subtitle(sub_url):
 
 
 class ITVBTCCIE(InfoExtractor):
-    _VALID_URL = r'https?://(?:www\.)?itv\.com/btcc/(?:[^/]+/)*(?P<id>[^/?#&]+)'
-    _TEST = {
-        'url': 'http://www.itv.com/btcc/races/btcc-2018-all-the-action-from-brands-hatch',
-        'info_dict': {
-            'id': 'btcc-2018-all-the-action-from-brands-hatch',
-            'title': 'BTCC 2018: All the action from Brands Hatch',
+    _VALID_URL = r'https?://(?:www\.)?itv\.com/btcc/(articles|races)/(?:[^/]+/)*(?P<id>[^/?#&]+)'
```
Comment (dstftw) [inline youtube_dl/extractor/itv.py:281]: Do not make the regex stricter.

File: youtube_dl/extractor/itv.py [hunk 3]
```diff
@@ -278,15 +278,25 @@ def extract_subtitle(sub_url):
...
-        'playlist_mincount': 9,
-    }
+        {
+            'url': 'http://www.itv.com/btcc/races/btcc-2018-all-the-action-from-brands-hatch',
+            'info_dict': {
+                'id': 'btcc-2018-all-the-action-from-brands-hatch',
+                'title': 'BTCC 2018: All the action from Brands Hatch',
+            },
+            'playlist_mincount': 9,
+        }
```
Comment (dstftw) [inline youtube_dl/extractor/itv.py:298]: Remove duplicate test.

File: youtube_dl/extractor/itv.py (hunk 1 above)
Comment (dstftw) [inline youtube_dl/extractor/itv.py:318]: Will not work if `id` is moved apart from `{`. Parse JSON instead.

**Comment by franhp** [comment]
Made some improvements based on code conventions, perhaps you could have another look?

**Comment by WolfganP** [comment]
I tested this PR and works ok with the IT//BTCC new pages organization.

**Pull request closed without merging**